magma/clock.py

```diff
@@ -87,16 +89,31 @@ def ClockInterface(has_enable=False, has_reset=False, has_ce=False,
     return args
 
 
+def do_clock_wire(port, clocktype, defnclk):
```

Comment:
I think eventually we should move these into a separate place like `wire_clocks.py` or even `auto_wire.py` since they're mostly general across type, but fine for now.